velox/expression/tests/CastExprTest.cpp

```diff
@@ -173,6 +173,160 @@ class CastExprTest : public functions::test::CastBaseTest {
+        {DecimalUtil::kShortDecimalMin,
+         DecimalUtil::kShortDecimalMin,
+         -3,
+         0,
+         55,
+         DecimalUtil::kShortDecimalMax,
+         DecimalUtil::kShortDecimalMax,
+         std::nullopt},
+        DECIMAL(18, 18));
+    testComplexCast(
+        "c0",
+        shortFlat,
+        makeNullableFlatVector<T>(
+            {-1,
+             // the same DecimalUtil::kShortDecimalMin conversion, checking
+             // floating point diff works on decimals
+             -0.999999999999999999,
+             -0.000000000000000003,
+             0,
+             0.000000000000000055,
+             // the same DecimalUtil::kShortDecimalMax conversion, checking
+             // floating point diff works on decimals
+             0.999999999999999999,
+             1,
+             std::nullopt}));
+
+    auto longFlat = makeNullableFlatVector<int128_t>(
+        {DecimalUtil::kLongDecimalMin,
+         0,
+         DecimalUtil::kLongDecimalMax,
+         HugeInt::build(0xffff, 0xffffffffffffffff),
+         std::nullopt},
+        DECIMAL(38, 5));
+    testComplexCast(
+        "c0",
+        longFlat,
+        makeNullableFlatVector<T>(
+            {-1e33, 0, 1e33, 1.2089258196146293E19, std::nullopt}));
+  }
+
+  template <TypeKind KIND>
+  void testDecimalToIntegralCastsOutOfBounds() {
+    using NativeType = typename TypeTraits<KIND>::NativeType;
+    VELOX_CHECK(!(std::is_same<int64_t, NativeType>::value));
+    const auto tooSmall =
+        static_cast<int64_t>(std::numeric_limits<int32_t>::min()) - 1;
+    const auto tooBig =
+        static_cast<int64_t>(std::numeric_limits<int32_t>::max()) + 1;
+
+    VELOX_ASSERT_THROW(
+        testComplexCast(
+            "c0",
+            makeFlatVector<int64_t>({0, tooSmall}, DECIMAL(10, 0)),
+            makeFlatVector<NativeType>(0, 0)),
+        fmt::format(
+            "Failed to cast from DECIMAL(10,0) to {}: -2147483649. Out of bounds.",
+            TypeTraits<KIND>::name));
+
+    VELOX_ASSERT_THROW(
+        testComplexCast(
+            "c0",
+            makeFlatVector<int128_t>({0, tooSmall}, DECIMAL(19, 0)),
+            makeFlatVector<NativeType>(0, 0)),
+        fmt::format(
+            "Failed to cast from DECIMAL(19,0) to {}: -2147483649. Out of bounds.",
+            TypeTraits<KIND>::name));
+
+    VELOX_ASSERT_THROW(
+        testComplexCast(
+            "c0",
+            makeFlatVector<int64_t>({0, tooBig}, DECIMAL(10, 0)),
+            makeFlatVector<NativeType>(0, 0)),
+        fmt::format(
+            "Failed to cast from DECIMAL(10,0) to {}: 2147483648. Out of bounds.",
+            TypeTraits<KIND>::name));
+
+    VELOX_ASSERT_THROW(
+        testComplexCast(
+            "c0",
+            makeFlatVector<int128_t>({0, tooBig}, DECIMAL(19, 0)),
+            makeFlatVector<NativeType>(0, 0)),
+        fmt::format(
+            "Failed to cast from DECIMAL(19,0) to {}: 2147483648. Out of bounds.",
```

Comment:
This message looks wrong. "to {}" should have something other than "{}", no?

Reply:
The name will replace in placeholder. It will be like Failed to cast from DECIMAL(19,0) to INTEGER: 2147483648. Out of bounds.